### DISCUSSION TITLE
How do I get the app to continue to run even after closing?

### DISCUSSION THREAD

**Discussion by Raj2032** [capawesome-team/capacitor-plugins#280] [Q&A]
I have tried this example so far:

https://github.com/capawesome-team/capacitor-plugins/tree/main/packages/background-task

```ts
import { App } from '@capacitor/app';
import { BackgroundTask } from '@capawesome/capacitor-background-task';

App.addListener('appStateChange', async ({ isActive }) => {
  if (isActive) {
    return;
  }
  // The app state has been changed to inactive.
  // Start the background task by calling `beforeExit`.
  const taskId = await BackgroundTask.beforeExit(async () => {
    // Run your code...
    // Finish the background task as soon as everything is done.
    BackgroundTask.finish({ taskId });
  });
});
```

However this works for only when the app is in background, not after when you have closed it. Is there a way to run your app even after you have closed it?

**Comment by robingenz** [comment]
No, if the user force closes your app then there is no way.

**Reply by Raj2032** [reply]
I didn't mean you got into app info and force stop, I meant like if you swipe off the app it still continues to run, think of a stop watch for example, most of those apps will still run even if you swipped off the app so that is what I mean?

**Reply by robingenz** [reply]
You can either move the app to the background or force close it (via swipe or via the settings app). If you force closes your app then there is no way.

**Reply by Raj2032** [reply]
@robingenz When you swipe the app off, it can still run in the background, like your timer app for example. I just mean if you swipe off the app I want it to still continue to run in the background (I know that it won't work if you force stop it via settings). Does any of your extensions support this?

**Reply by robingenz** [reply]
The only option I can think of is a separate Foreground service. But even this is terminated on my Android device (OnePlus) when I swipe the app away.
